cluster/aws/config-test.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ZONE=${KUBE_AWS_ZONE:-us-west-2a}
+
+MASTER_SIZE=${MASTER_SIZE:-}
+NODE_SIZE=${NODE_SIZE:-}
+NUM_NODES=${NUM_NODES:-2}
+
+# Dynamically set node sizes so that Heapster has enough space to run
+if [[ -z ${NODE_SIZE} ]]; then
+  if (( ${NUM_NODES} < 50 )); then
+    NODE_SIZE="t2.micro"
+  elif (( ${NUM_NODES} < 150 )); then
+    NODE_SIZE="t2.small"
+  else
+    NODE_SIZE="t2.medium"
+  fi
+fi
+
+# Dynamically set the master size by the number of nodes, these are guesses
+# TODO: gather some data
+if [[ -z ${MASTER_SIZE} ]]; then
+  if (( ${NUM_NODES} < 150 )); then
+    MASTER_SIZE="m3.medium"
+  else
+    MASTER_SIZE="m3.large"
+  fi
+fi
+
+
+# Because regions are globally named, we want to create in a single region; default to us-east-1
+AWS_S3_REGION=${AWS_S3_REGION:-us-east-1}
+
+# Which docker storage mechanism to use.
+DOCKER_STORAGE=${DOCKER_STORAGE:-aufs}
+
+# Extra docker options for nodes.
+EXTRA_DOCKER_OPTS="${EXTRA_DOCKER_OPTS:-}"
+
+INSTANCE_PREFIX="${KUBE_AWS_INSTANCE_PREFIX:-e2e-test-${USER}}"
+CONFIG_CONTEXT="${KUBE_CONFIG_CONTEXT:-aws_${INSTANCE_PREFIX}}"
+CLUSTER_ID=${INSTANCE_PREFIX}
+VPC_NAME=${VPC_NAME:-kubernetes-vpc}
+AWS_SSH_KEY=${AWS_SSH_KEY:-$HOME/.ssh/kube_aws_rsa}
+
+LOG="/dev/null"
+
+MASTER_DISK_TYPE="${MASTER_DISK_TYPE:-gp2}"
```

Comment:
There seems many duplicated variables. Can those be deduped?

Reply:
this is for e2e test.